### PR TITLE
kubectl explain displays path of kind instead of "Object"

### LINKED ISSUE #80386: kubectl explain display Object instead of the name of the Kind
**What happened**:

The command `kubectl explain pod.spec` displays " Object " for the resource spec.

**What you expected to happen**:

I would expect it displays the real name of the Kind, so I can use it in a Go program, and possibly with the full path, so I can use the path for the Go import.

**How to reproduce it (as minimally and precisely as possible)**:

kubectl explain pod.spec

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

The "kubectl explain" command displays "Object" for non primitive fields, and it's not usable when using the output of explain to write Go code.

I propose to display the full path of the field, so the Go developer can use it to find the Go import and the name of the structure.

Example:

```
$ kubectl explain pod.spec

KIND:     Pod
VERSION:  v1

RESOURCE: spec <io.k8s.api.core.v1.PodSpec>
[...]
   dnsConfig    <io.k8s.api.core.v1.PodDNSConfig>
     Specifies the DNS parameters of a pod. Parameters specified here will be
     merged to the generated DNS configuration based on DNSPolicy.
```

With this output, the developer can easily find the import "k8s.io/api/core/v1" and the name of the structs PodSpec and PodDNSConfig.

**Which issue(s) this PR fixes**:
Fixes #80386

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
kubectl explain displays path of kind instead of "Object"
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by apelisse** [comment]
That looks reasonable to me! Thanks for doing that!

File: test/e2e/apimachinery/crd_publish_openapi.go
Comment (jellonek) [inline test/e2e/apimachinery/crd_publish_openapi.go]: Could this test expect more precise match in this (and below) places?

Comment (feloy) [reply test/e2e/apimachinery/crd_publish_openapi.go]: @jellonek Yes, you are right. I fixed the tests. Thanks

**Review by jellonek — APPROVED** [review]
lgtm

**Comment by feloy** [comment]
> lgtm

Sorry @jellonek I made a wrong manipulation by requesting a new review

**Comment by jellonek** [comment]
np. This PR still requires approval from someone from @kubernetes/sig-cli-maintainers group

File: test/e2e/apimachinery/crd_publish_openapi.go
Comment (liggitt) [inline test/e2e/apimachinery/crd_publish_openapi.go]: is this actually better/more desireable?

File: staging/src/k8s.io/kubectl/pkg/explain/typename.go
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/explain/typename.go]: `GetPath().String()` is more readable in some contexts, and way less readable in others... (e.g. https://github.com/kubernetes/kubernetes/pull/80387/files#r328156170)

**Comment by liggitt** [comment]
> it's not usable when using the output of explain to write Go code

I would expect the type definitions to be used when writing go code, not `kubectl explain` output.

File: test/e2e/apimachinery/crd_publish_openapi.go
Comment (feloy) [reply test/e2e/apimachinery/crd_publish_openapi.go]: In this case no, but I consider this a very specific case

**Pull request closed without merging**
